### ISSUE TITLE
ivar can be negative in resample_flux

### ISSUE THREAD

**Issue by londumas** [desihub/desispec#511]
When re-sampling data with `desispec.interpolation.resample_flux` the output ivar can be negative.
Here is an example with the data

```
import scipy as sp
from desispec.interpolation import resample_flux

### Get data
spec = sp.loadtxt('issue_spectrum.txt')
twave = spec[:,0]
tflux = spec[:,1]
tivar = spec[:,2]
print('INFO: has some neg?: ', sp.any(tivar<0.))

### Resample
CRVAL1 = 2.6534
CDELT1 = 0.0001
NBLL   = 13637
wave   = 10**(CRVAL1+sp.arange(NBLL)*CDELT1)

flux, ivar = resample_flux(wave, twave, tflux, tivar)
print('INFO: has some neg?: ', sp.any(ivar<0.))
if sp.any(ivar<0.):
    w = ivar<0.
    print(wave[w])
    print(flux[w])
    print(ivar[w])
```
[issue_spectrum.txt](https://github.com/desihub/desispec/files/1741708/issue_spectrum.txt)

**Comment by londumas** [comment]
The issue comes from the following line:
https://github.com/desihub/desispec/blob/f32955b95534cb1864a425dcb0ec8fb9c5ba9c74/py/desispec/interpolation.py#L195 that gives a negative ivar for this spectrum

**Comment by julienguy** [comment]
That's bad ... thanks for this Helion, I am looking at it.

**Comment by julienguy** [comment]
Addressed in PR #646.
